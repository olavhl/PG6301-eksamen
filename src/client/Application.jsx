import React from "react";
import { BrowserRouter, Link } from "react-router-dom";
import { Route, Switch } from "react-router";

export function Application() {
  return (
    <BrowserRouter>
      <header>
        <Link to={"/"}>
          <button>Home</button>
        </Link>
      </header>
      <Switch>
        <Route exact path={"/login"}>
          <h1>Login</h1>
        </Route>
        <Route path={"/login/callback"}>
          <h1>Login Callback</h1>
        </Route>
        <Route exact path={"/"}>
          <h1>Welcome to the chatting page</h1>
          <ul>
            <li>
              <Link to={"/login"}>Login</Link>
            </li>
          </ul>
        </Route>
        <Route>
          <h1>Page not found...</h1>
        </Route>
      </Switch>
    </BrowserRouter>
  );
}
