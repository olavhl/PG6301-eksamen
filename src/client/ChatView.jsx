import React, { useState } from "react";

export function ChatView({ onSendMessage, chat }) {
  const [message, setMessage] = useState("");
  // Feedback
  const [feedBack, setFeedBack] = useState("");

  function handleSubmitChat(e) {
    e.preventDefault();

    if (message === "") {
      setFeedBack("Write the message you want to send!");
      return;
    }

    onSendMessage(message);
    setMessage("");
  }

  return (
    <div className={"chat"}>
      <header>
        <h1>Chat</h1>
      </header>
      <main>
        {chat?.map((message, index) => (
          <div key={`${message}__${index}`}>{message}</div>
        ))}
      </main>
      {feedBack && <div className={"feedback"}>{feedBack}</div>}
      <footer>
        <form onSubmit={handleSubmitChat}>
          <input
            type="text"
            autoFocus={true}
            value={message}
            onChange={(e) => setMessage(e.target.value)}
          />
          <button>Send</button>
        </form>
      </footer>
    </div>
  );
}
