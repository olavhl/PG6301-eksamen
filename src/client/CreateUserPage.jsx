import React, { useState } from "react";
import { InputField } from "./components/InputField";

export function CreateUserPage() {
  const [firstName, setFirstName] = useState("");
  const [lastName, setLastName] = useState("");
  const [email, setEmail] = useState("");

  async function handleSubmit(e) {
    e.preventDefault();
  }

  return (
    <div>
      <h1>Create user</h1>
      <form onSubmit={handleSubmit}>
        <InputField
          value={firstName}
          label={"First name"}
          onChangeValue={setFirstName}
        />
        <InputField
          value={lastName}
          label={"Last name"}
          onChangeValue={setLastName}
        />
        <InputField value={email} label={"Email"} onChangeValue={setEmail} />
        <button>Submit</button>
      </form>
    </div>
  );
}
