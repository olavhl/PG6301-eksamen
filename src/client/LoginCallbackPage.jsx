import React, { useEffect } from "react";
import { useHistory } from "react-router";

export function LoginCallbackPage({ onAccessToken }) {
  const hash = Object.fromEntries(
    new URLSearchParams(window.location.hash.substring(1))
  );

  const history = useHistory();

  useEffect(() => {
    const { access_token } = hash;
    onAccessToken(access_token);
    history.push("/");
  }, []);

  return <h1>Login Callback</h1>;
}
