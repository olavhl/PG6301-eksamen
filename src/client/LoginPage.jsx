import React from "react";
import { fetchJSON } from "./lib/http";

export function LoginPage({ identityProvider }) {
  const { discoveryURL, client_id } = identityProvider;

  async function handleLogin() {
    const { authorization_endpoint } = await fetchJSON(discoveryURL);
    const params = {
      client_id,
      response_type: "token",
      scope: "openid email profile",
      redirect_uri: window.location.origin + "/login/callback",
    };

    window.location.href =
      authorization_endpoint + "?" + new URLSearchParams(params);
  }

  return (
    <div>
      <h1>Login</h1>
      <button onClick={handleLogin}>Log in</button>
    </div>
  );
}
