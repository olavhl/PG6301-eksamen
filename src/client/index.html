<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <title>Messaging System</title>
    <link rel="stylesheet" href="style.css" type="text/css">
  </head>
  <body>
    <div id="app"></div>
  </body>
  <script src="index.jsx"></script>
</html>
