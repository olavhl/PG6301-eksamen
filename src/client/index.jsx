import React from 'react';
import ReactDOM from 'react-dom';

function Application() {
    return <h1>Hello</h1>;
}

ReactDOM.render(<Application/>, document.getElementById("app"));