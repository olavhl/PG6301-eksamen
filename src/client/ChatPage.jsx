import React, { useEffect, useState } from "react";
import { ChatView } from "./ChatView";

export function ChatPage() {
  const [chat, setChat] = useState([]);
  const [ws, setWs] = useState();

  useEffect(() => {
    const ws = new WebSocket("ws://localhost:3000/chat");
    ws.onopen = (event) => {
      console.log("opened", event);
    };

    ws.onmessage = (event) => {
      console.log("message", event);
      setChat((chat) => [...chat, event.data]);
    };

    ws.onclose = (event) => {
      console.log("close", event);
    };

    setWs(ws);
  }, []);

  return <ChatView chat={chat} onSendMessage={(message) => ws.send(message)} />;
}
