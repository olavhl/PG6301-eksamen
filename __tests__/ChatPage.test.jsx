import React from "react";
import ReactDOM from "react-dom";
import { ChatPage } from "../src/client/ChatPage";

describe("chat page", () => {
  it("shows chat page", async () => {
    const container = document.createElement("div");
    ReactDOM.render(<ChatPage />, container);

    expect(container.querySelector("h1").textContent).toEqual("Chat");
    expect(container.innerHTML).toMatchSnapshot();
  });
});
