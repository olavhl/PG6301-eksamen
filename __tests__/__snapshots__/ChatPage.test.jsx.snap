// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat page shows chat page 1`] = `"<div class=\\"chat\\"><header><h1>Chat</h1></header><h3>Write you first message</h3><main></main><footer><form><input type=\\"text\\" value=\\"\\"><button>Send</button></form></footer></div>"`;
